Tests/Algorithm/AlgorithmRegisterIndicatorTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using NUnit.Framework;
+using Python.Runtime;
+using QuantConnect.Algorithm;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Indicators;
+using QuantConnect.Tests.Indicators;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Tests.Algorithm
+{
+    [TestFixture]
+    public class AlgorithmRegisterIndicatorTests
+    {
+        private Symbol _spy;
+        private QCAlgorithm _algorithm;
+        private IEnumerable<IGrouping<Type, Type>> _indicatorTestsGroups;
+
+        [SetUp]
+        public void Setup()
+        {
+            _algorithm = new QCAlgorithm();
+            _spy = _algorithm.AddEquity("SPY").Symbol;
+
+            _indicatorTestsGroups =
+                from type in GetType().Assembly.GetTypes()
+                where type.IsPublic && !type.IsAbstract
+                where
+                   typeof(CommonIndicatorTests<TradeBar>).IsAssignableFrom(type) ||
+                   typeof(CommonIndicatorTests<IBaseDataBar>).IsAssignableFrom(type) ||
+                   typeof(CommonIndicatorTests<IndicatorDataPoint>).IsAssignableFrom(type)
+                group type by type.BaseType.GetGenericArguments().FirstOrDefault();
```

Comment:
I love seeing you get more and more comfortable with the reflection libraries. This is great.

Hmm... which indicators should I test... I know! I'll test **all** of them :D